### PR TITLE
RHMAP 6004

### PR DESCRIPTION
**Pull request by odra**
# Motivation

Create new  FHC commands to export, list and download app data.

This is not 100% ready yet since it needs the fh-supercore APIs (using dummy data  right now).

## Command list

**Export data for a given app**
`fhc appdata export --env=<env> --app=<appid>`

**List export history**
`fhc appdata list`

**Download a previous export data**
`fhc appdata download --job=<jobid>`

### CODE REVIEW

File: lib/cmd/fh3/appdata.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+    return cb(null, data);
+  }
+}
+
+function exportAppData(args, cb) {
+  var app = args.app;
+  var env = args.env;
+  var path = 'api/v2/appdata/' + env + '/' + app + '/data/export';
+
```
Comment (fh-qube) [inline lib/cmd/fh3/appdata.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)

File: lib/cmd/fh3/appdata.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  var env = args.env;
+  var path = 'api/v2/appdata/' + env + '/' + app + '/data/export';
+  
+  var rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+  });
+
+  var question = 'Do you want to stop the application first?' +
+    'The application data might be outdated if you choose not to. (Y/N):';
```
Comment (fh-qube) [inline lib/cmd/fh3/appdata.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Comment by fh-automatron** [comment]
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:

* The code changes conform to the contributing & code style guides for this component (if they exist)
* A component owner should review your change (we've tagged these above)
* Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
* There should be an associated fhcap PR with this change - please provide a link to this
* If updating server component this PR should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
* Ensure JIRA is up to date before & after merge (e.g. deployment notes, tester notes, release notes, fix version)
* Any SonarQube issues should be addressed where applicable (use your own judgement)

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.

In order to update fhcap with your component change please execute:
`fhcap component update fh-fhc 2.7.3 308`

File: lib/cmd/fh3/appdata.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  var env = args.env;
+  var path = 'api/v2/appdata/' + env + '/' + app + '/data/export';
+
+  var rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+  });
+
+  var question = 'Do you want to stop the application first?' +
+  'The application data might be outdated if you choose not to. (Y/N):';
```
Comment (fh-qube) [inline lib/cmd/fh3/appdata.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: lib/cmd/fh3/appdata.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+}
+
+function shouldStopApp(text) {
+  if (!text) {
+    return true;
+  }
+
+  var _text = text.toLowerCase();
+
+  if (_text == 'y' || _text == 'yes') {
```
Comment (fh-qube) [inline lib/cmd/fh3/appdata.js]: BLOCKER Replace "==" with "===". [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AEqEqEq)
BLOCKER Replace "==" with "===". [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AEqEqEq)

File: lib/cmd/fh3/appdata.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+  var rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+  });
+
+  var question = [
+    'Do you want to stop the application first?',
+    'The application data might be outdated if you choose not to. (Y/N):'
+  ].join('');
```
Comment (fh-qube) [inline lib/cmd/fh3/appdata.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)

**Comment by fh-qube** [comment]
SonarQube analysis reported 3 issues:
* BLOCKER 2 blocker
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by pb82** [comment]
At the moment it allows me to execute for example `fhc appdata list` without any additional arguments. But we always require `environment` and `appId`. They should be mandatory.

File: lib/cmd/fh3/appdata.js [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+
+  if (!fn) {
+    return unknown("Invalid service action " + action, cb);
+  }
+
+  return fn(argv, cb);
+}
+
+function listAppData(args, cb) {
+  var app = args.app;
```
Comment (pb82) [inline lib/cmd/fh3/appdata.js]: Maybe call it `addId` to make clear that it's about the app id (and not it's label)

File: lib/cmd/fh3/appdata.js [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+    }
+
+    return cb(null, 'Job ID: ' + result);
+  }
+}
+
+function downloadAppData(args, cb) {
+  var app = args.app;
+  var env = args.env;
+  var job = args.job;
```
Comment (pb82) [inline lib/cmd/fh3/appdata.js]: Better call it `jobId`

**Comment by odra** [comment]
see #212

**Pull request closed without merging**